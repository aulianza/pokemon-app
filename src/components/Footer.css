footer {
    padding: .1rem;
    background-color: #2d74b9;
    color: #fff;
}

footer a {
    color: #fff;
}