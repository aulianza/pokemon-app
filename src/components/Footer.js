import React from "react";
import styled from "@emotion/styled";

const Footer = () => {
    return (
        <StyledFooter>
            <footer>
                <div>
                    <p>
                        &#169; 2021 by{" "}
                        <a href="https://aulianza.id" target="_blank" rel="noreferrer">
                            aulianza
                        </a>
                    </p>
                </div>
            </footer>
        </StyledFooter>
    );
};

export default Footer;

const StyledFooter = styled.div`
    padding: 0.1rem;
    background-color: #2d74b9;
    color: #fff;

    a {
        color: #fff;
    }
`;
