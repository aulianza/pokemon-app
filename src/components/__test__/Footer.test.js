import Footer from "../Footer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from "react-router-dom";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("footer-component, snapshot", () => {
    act(() => {
        render(
            <Router>
                <Footer />
            </Router>,
            container
        );
    });
    expect(container).toMatchSnapshot();
});
