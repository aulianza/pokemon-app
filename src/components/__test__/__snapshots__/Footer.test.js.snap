// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer-component, snapshot 1`] = `
<div>
  <footer>
    <div>
      <p>
        Â© 2021 by 
        <a
          href="https://aulianza.id"
          rel="noreferrer"
          target="_blank"
        >
          aulianza
        </a>
      </p>
    </div>
  </footer>
</div>
`;
